Docs12227/azure logs troubleshooting (#33113)
DataDog/documentation: content/en/integrations/guide/azure-advanced-configuration.md

- Failed

If you see a status of `query_failed` you need to enable the Resource Health provider in Azure.
{{% /collapse-content %}} 

## Azure integration troubleshooting

> {{% collapse-content title="Azure integration troubleshooting" level="h4" expanded=false id="troubleshooting-find-your-tenant-name" %}}

Reviewer: ```suggestion
{{% collapse-content title="Find your tenant name" level="h4" expanded=false id="troubleshooting-find-your-tenant-name" %}}
```

#### Reconcile the discrepancy

1. Graph the metric in [Azure Monitor Metrics Explorer][18], or by going to the resource in Azure and clicking **Monitoring** and then **Metrics** in the left panel.
2. Graph the metric in the [Datadog Metrics Explorer][31].
3. Confirm that the query in Azure is scoped identically to the query in Datadog:
   - Any dimensions used in the Azure metric query should match tags used in the Datadog metric query
   - The primary aggregation type used in the query should match the Datadog [space aggregator][28]
   - The time frame should match the time frame in the Datadog Metric Explorer
4. Hover over a datapoint on the graph to display the timestamp and value.

{{< img src="integrations/guide/azure_advanced_configuration/azure_metric_explorer.png" alt="The Azure Metrics Explorer with the cursor hovering over a point in the graph, and the metric value and timestamp highlighted" >}}

5. Find the same point in time in the Datadog graph and compare the values. If the values are equal, the original discrepancy was due to differences in either time or space aggregation between the two graphs.
{{% /collapse-content %}} 

### Log collection troubleshooting

> Use the information in this section to troubleshoot log forwarding issues. If your issue isn't described in the [general troubleshooting section](#general-troubleshooting-applicable-to-all-setup-methods), click the link below that corresponds to your setup method for more specific troubleshooting steps.

Reviewer: ```suggestion
To troubleshoot log forwarding issues, begin with the [general troubleshooting section](#general-troubleshooting-applicable-to-all-setup-methods). If those steps do not resolve your issue, look at the following setup-specific guidance. 
```

Exceeding the daily log quota:
: Ensure that you have not exceeded your [daily quota][20] for log retention.

Excluding logs with exclusion filters:
: Ensure that your logs are not excluded by [exclusion filters][38].

Incorrect syntax used to search logs:
: Ensure that you used the correct [search syntax][34] and that your search contains the correct attributes or tags.

Diagnostic setting not present:
: Ensure that a [diagnostic setting][8] has been added to the resources, Activity Logs, and Entra ID logs you want to send to Datadog.

**Note**: You can search for Entra ID logs with the query `source:azure.activedirectory`.

#### Automated log forwarding troubleshooting

{{% collapse-content title="Check for naming conflicts" level="h4" expanded=false id="troubleshooting-logs-naming-conflicts" %}}
If you have Azure resources with the same resource name as one of the default parameters, it can lead to naming conflicts. Azure does not allow resources to share resource names within an individual subscription. Datadog recommends renaming the default parameter with a unique name that does not already exist within your environment.

> **Note**: If you are rerunning the template due to this failure, Datadog also recommends removing the entire resource group to create a fresh deployment. 

Reviewer: ```suggestion
**Note**: If you are rerunning the template due to this failure, Datadog recommends also removing the entire resource group to create a fresh deployment. 
```
